public/app/features/dashboard/api/v2.ts

```diff
@@ -31,9 +32,23 @@ export class K8sDashboardV2APIStub implements DashboardAPI<DashboardWithAccessIn
   async getDashboardDTO(uid: string, params?: UrlQueryMap) {
     const dashboard = await this.client.subresource<DashboardWithAccessInfo<DashboardV2Spec>>(uid, 'dto');
 
-    // For dev purposes only, the conversion should and will happen in the API. This is just to stub v2 api responses.
-    const result = ResponseTransformers.ensureV2Response(dashboard);
+    let result: DashboardWithAccessInfo<DashboardV2Spec>;
 
+    // TODO: For dev purposes only, the conversion should and will happen in the API. This is just to stub v2 api responses.
+    // if the dashboard is a resource, meaning is in k8s, and is not a v2 spec, then convert it to a v2 spec
+    if (isDashboardResource(dashboard) && !isDashboardV2Spec(dashboard)) {
+      //this mean the dashboard is in v0 format
+      const dto = dashboard as unknown as DashboardWithAccessInfo<DashboardDataDTO>;
+      // convert to v1 schema
+      const v1Response = ResponseTransformers.ensureV1ResponseFromV0(dto);
```

Comment:
Don't understand why we need to transform to v1 (it's already v1 wrapped in k8s format). can't we simply `ensureV2` on it?